packages/block-library/src/navigation/edit/index.js

```diff
@@ -95,13 +95,49 @@ function Navigation( {
 
 	const ref = attributes.ref;
 
-	const setRef = ( postId ) => {
-		setAttributes( { ref: postId } );
+	const setRef = ( postSlug ) => {
+		setAttributes( { ref: postSlug } );
 	};
 
 	const recursionId = `navigationMenu/${ ref }`;
 	const hasAlreadyRendered = useHasRecursion( recursionId );
 
+	const generatedTemplatePartHierarchySlug = useSelect(
```

Comment:
Should we try to 1st move from id to slug, _then_ implement the better slug formation?

Reply:
Yes we could do that if we want to further simplify this PR.

Reply:
My current thought is that we will want to create another endpoint which handles the "lookup" for fallbacks. The `get_item` endpoint should be looking up Posts based on a "known" reference.

We would create a `lookup` endpoint which would receive a slug and then retrieve the post which most closely matches the template hierarchy encoded within the slug.